SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialWorkerFlags/SpatialWorkerFlagsTest.cpp

```diff
@@ -51,39 +51,160 @@ SPATIALWORKERFLAGS_TEST(GIVEN_a_flagUpdate_op_WHEN_removing_a_worker_flag_THEN_f
+	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetAnyFlagUpdated);
 
 	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
-	SpatialWorkerFlags->BindToOnWorkerFlagsUpdated(WorkerFlagDelegate);
+	SpatialWorkerFlags->RegisterAnyFlagUpdatedCallback(WorkerFlagDelegate);
+
+	// Add test flag
+	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp("test", "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
+
+	// Add test flag 2
+	Worker_FlagUpdateOp OpAddFlag2 = CreateWorkerFlagUpdateOp("test2", "20");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag2);
+
+	TestTrue("Delegate Function was called twice", SpyObj->GetTimesFlagUpdated() == 2);
+
+	return true;
+}
+
+SPATIALWORKERFLAGS_TEST(GIVEN_a_registered_any_flag_update_delegate_WHEN_unregistering_the_delegate_THEN_delegate_is_not_invoked)
+{
+	UWorkerFlagsTestSpyObject* SpyObj = NewObject<UWorkerFlagsTestSpyObject>();
+	FOnAnyWorkerFlagUpdatedBP WorkerFlagDelegate;
+	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetAnyFlagUpdated);
 
+	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
+	SpatialWorkerFlags->RegisterAnyFlagUpdatedCallback(WorkerFlagDelegate);
 	// Add test flag
 	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp("test", "10");
 	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
 
 	TestTrue("Delegate Function was called", SpyObj->GetTimesFlagUpdated() == 1);
 
+	SpatialWorkerFlags->UnregisterAnyFlagUpdatedCallback(WorkerFlagDelegate);
+
+	// Update test flag
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
+
+	TestTrue("Delegate Function was called only once", SpyObj->GetTimesFlagUpdated() == 1);
+
+	return true;
+}
+
+// any flag invoke
+
+SPATIALWORKERFLAGS_TEST(GIVEN_an_updated_flag_WHEN_registering_and_invoking_an_any_flag_update_delegate_THEN_delegate_is_invoked)
+{
+	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
+
+	// Add test flag
+	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp("test", "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
+
+	// register callback
+	UWorkerFlagsTestSpyObject* SpyObj = NewObject<UWorkerFlagsTestSpyObject>();
+	FOnAnyWorkerFlagUpdatedBP WorkerFlagDelegate;
+	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetAnyFlagUpdated);
+
+	SpatialWorkerFlags->RegisterAndInvokeAnyFlagUpdatedCallback(WorkerFlagDelegate);
+	// should invoke immediately
+	TestTrue("Delegate Function was called once", SpyObj->GetTimesFlagUpdated() == 1);
+
+	// Add test flag 2
+	Worker_FlagUpdateOp OpAddFlag2 = CreateWorkerFlagUpdateOp("test2", "20");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag2);
+
+	TestTrue("Delegate Function was called twice", SpyObj->GetTimesFlagUpdated() == 2);
+
+	return true;
+}
+
+SPATIALWORKERFLAGS_TEST(GIVEN_no_flags_WHEN_registering_and_invoking_an_any_flag_update_delegate_THEN_delegate_is_not_invoked)
+{
+	// register callback
```

Comment:
Very minor nit, let's go with consistency in comments and start with caps, throughout the class.